qualtran/bloqs/reflections/reflection_using_prepare.py

```diff
@@ -187,7 +187,9 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
         if self.global_phase != 1:
             phase_op: Bloq = GlobalPhase.from_coefficient(self.global_phase, eps=self.eps)
             if self.control_val is not None:
-                phase_op = phase_op.controlled(ctrl_spec=CtrlSpec(cvs=self.control_val))
+                phase_op = phase_op.controlled()
+                if self.control_val == 0:
+                    costs[XGate()] = 2
```

Comment:
should this be `+= 2`? (as there's already an `XGate()` in L186)

Reply:
that's guarded by `if control_val is None` and this line test `if control_val == 0`